drivers/sensor/ds18b20/ds18b20.c

```diff
@@ -0,0 +1,259 @@
+	}
+
+	status = ds18b20_temperature_convert(dev);
+	if (status < 0) {
+		LOG_DBG("W1 fetch error");
+		return status;
+	}
+	k_msleep(measure_wait_ms[DS18B20_RESOLUTION_INDEX(cfg->resolution)]);
+	return ds18b20_read_scratchpad(dev, &data->scratchpad);
+}
+
+static int ds18b20_channel_get(const struct device *dev,
+			       enum sensor_channel chan,
+			       struct sensor_value *val)
+{
+	struct ds18b20_data *data = dev->data;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	ds18b20_temperature_from_raw((uint8_t *)&data->scratchpad.temp, val);
+	return 0;
+}
+
+static int ds18b20_configure(const struct device *dev)
+{
+	const struct ds18b20_config *cfg = dev->config;
+	struct ds18b20_data *data = dev->data;
+
+	if (w1_reset_bus(cfg->bus) <= 0) {
+		LOG_ERR("No 1-Wire devices connected");
+		return -ENODEV;
+	}
+
+	/* In single drop configurations the rom can be read from device */
+	if (w1_get_peripheral_count(cfg->bus) == 1) {
+		if (w1_rom_to_uint64(&data->config.rom_id) == 0ULL) {
+			(void)w1_read_rom(cfg->bus, &data->config.rom_id);
+		}
+	} else if (w1_rom_to_uint64(&data->config.rom_id) == 0ULL) {
+		LOG_DBG("nr: %d", w1_get_peripheral_count(cfg->bus));
+		LOG_ERR("ROM required, because multiple devices are on the bus");
+		return -EINVAL;
+	}
+
+	if ((cfg->family != 0) && (cfg->family != data->config.rom_id.family)) {
+		LOG_ERR("Found 1-Wire device is not a DS18B20");
+		return -EINVAL;
+	}
+
+	ds18b20_read_power_supply(dev);
+	LOG_DBG("Using external power supply: %d", data->power_supply);
+
+	/* write defaults */
+	ds18b20_set_resolution(dev, cfg->resolution);
+	ds18b20_write_scratchpad(dev, data->scratchpad);
+	LOG_DBG("Init DS18B20: ROM=%016llx\n",
+		w1_rom_to_uint64(&data->config.rom_id));
+
+	return 0;
+}
+
+int ds18b20_attr_set(const struct device *dev, enum sensor_channel chan,
+		     enum sensor_attribute attr, const struct sensor_value *thr)
+{
+	struct ds18b20_data *data = dev->data;
+
+	if ((enum sensor_attribute_w1)attr != SENSOR_ATTR_W1_ROM_ID) {
+		return -ENOTSUP;
+	}
+
+	data->lazy_loaded = false;
+	w1_sensor_value_to_rom(thr, &data->config.rom_id);
+	return 0;
+}
+
+static const struct sensor_driver_api ds18b20_driver_api = {
+	.attr_set = ds18b20_attr_set,
+	.sample_fetch = ds18b20_sample_fetch,
+	.channel_get = ds18b20_channel_get,
+};
+
+static int ds18b20_init(const struct device *dev)
+{
+	const struct ds18b20_config *cfg = dev->config;
+	struct ds18b20_data *data = dev->data;
+
+	if (device_is_ready(cfg->bus) == 0) {
+		LOG_DBG("w1 bus is not ready");
+		return -ENODEV;
+	}
+
+	w1_uint64_to_rom(0ULL, &data->config.rom_id);
+	data->lazy_loaded = false;
+	/* in multi drop configurations the rom is need, but is not set during
+	 * driver initialization, therefore do lazy initialization in all cases.
+	 */
+
+	return 0;
+}
+
+#define DS18B20_CONFIG_INIT(inst)					\
+	{								\
+		.bus = DEVICE_DT_GET(DT_INST_BUS(inst)),		\
+		.family = (uint8_t)DT_INST_PROP(inst, family_code),	\
```

Comment:
You cannot safely use DT_INST_PROP on a property without `required: true`. Should this be using `DT_INST_PROP_OR` instead? Or maybe the maxim,dsb18b20.yaml file is missing a `required: true` for this property?